File: frontend/refactor.js
const fs = require('fs');
const path = require('path');

const srcDir = 'd:\\Gym-App\\Fit-Ledger\\frontend\\src';

function walk(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(function(file) {
        file = path.join(dir, file);
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) { 
            results = results.concat(walk(file));
        } else { 
            results.push(file);
        }
    });
    return results;
}

const files = walk(srcDir);

for (const file of files) {
    if (!file.endsWith('.tsx') && !file.endsWith('.ts')) continue;
    if (file.includes('contexts\\ThemeContext') || 
        file.includes('navigation\\index.tsx') || 
        file.includes('theme\\index.ts') || 
        file.includes('screens\\profile\\page.tsx')) {
        continue;
    }

    let content = fs.readFileSync(file, 'utf8');
    if (!content.includes('import { theme }') && !content.includes('import theme ')) {
        continue;
    }

    console.log("Processing " + file);

    // Replace import
    content = content.replace(/import\s+\{\s*theme\s*\}\s+from\s+['"](.*?)theme['"]/g, "import { useTheme } from '$1contexts/ThemeContext'");
    content = content.replace(/import\s+theme\s+from\s+['"](.*?)theme['"]/g, "import { useTheme } from '$1contexts/ThemeContext'");

    // Replace StyleSheet.create
    content = content.replace("const styles = StyleSheet.create({", "const getStyles = (theme: any) => StyleSheet.create({");

    // Insert hook
    const hookStr = "\n  const { theme } = useTheme();\n  const styles = getStyles(theme);";
    
    // Pattern 1: export default function Name(props) {
    let matched = false;
    content = content.replace(/export default function \w+\([^)]*\)\s*\{/g, (match) => {
        matched = true;
        if (content.includes("const { theme } = useTheme()")) return match;
        return match + hookStr;
    });

    if (!matched) {
        content = content.replace(/export (?:const|function) \w+\s*=?\s*\([^)]*\)\s*(?:=>\s*)?\{/g, (match) => {
            if (content.includes("const { theme } = useTheme()")) return match;
            return match + hookStr;
        });
    }

    fs.writeFileSync(file, content, 'utf8');
}
console.log("Done");

File: frontend/refactor.py
import os
import re

src_dir = r"d:\Gym-App\Fit-Ledger\frontend\src"

for root, _, files in os.walk(src_dir):
    for file in files:
        if file.endswith('.tsx') or file.endswith('.ts'):
            path = os.path.join(root, file)
            with open(path, 'r', encoding='utf-8') as f:
                content = f.read()
            
            # Skip Profile, App.tsx, navigation, and theme files
            if "contexts\\ThemeContext" in path or "navigation\\index.tsx" in path or "theme\\index.ts" in path or "screens\\profile\\page.tsx" in path:
                continue
                
            if "import { theme }" not in content and "import theme " not in content:
                continue
                
            print(f"Processing {path}")
            
            # Replace import
            content = re.sub(
                r"import\s+\{\s*theme\s*\}\s+from\s+['\"](.*?)theme['\"]",
                r"import { useTheme } from '\1contexts/ThemeContext'",
                content
            )
            content = re.sub(
                r"import\s+theme\s+from\s+['\"](.*?)theme['\"]",
                r"import { useTheme } from '\1contexts/ThemeContext'",
                content
            )
            
            # Replace StyleSheet.create
            content = content.replace("const styles = StyleSheet.create({", "const getStyles = (theme: any) => StyleSheet.create({")
            
            # Insert hook into functional components
            # We look for export default function Name(props) {
            # or export function Name(props) {
            # or const Name = (props) => {
            
            def insert_hook(match):
                prefix = match.group(0)
                if "useTheme()" in content:
                    return prefix # already has it somewhere? just return
                return prefix + "\n  const { theme } = useTheme();\n  const styles = getStyles(theme);"
                
            content = re.sub(r"export default function \w+\([^)]*\)\s*\{", insert_hook, content)
            
            # For components like `const CustomAlert = () => {`
            if "export default function" not in content:
                content = re.sub(r"export (?:const|function) \w+\s*=?\s*\([^)]*\)\s*(?:=>\s*)?\{", insert_hook, content)
            
            with open(path, 'w', encoding='utf-8') as f:
                f.write(content)
                
print("Done")
